Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/FAQFragment.kt

```diff
@@ -0,0 +1,198 @@
+package com.mentalmachines.droidcon_boston.views
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.text.TextUtils
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseExpandableListAdapter
+import android.widget.ExpandableListView
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import com.bumptech.glide.Glide
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.ValueEventListener
+import com.mentalmachines.droidcon_boston.R
+import com.mentalmachines.droidcon_boston.data.FirebaseDatabase.FaqEvent
+import com.mentalmachines.droidcon_boston.data.FirebaseDatabase.FaqEvent.Answer
+import com.mentalmachines.droidcon_boston.firebase.FirebaseHelper
+import java.util.ArrayList
+import java.util.HashMap
+
+class FAQFragment : Fragment() {
+
+    private val firebaseHelper = FirebaseHelper.instance
+
+    private lateinit var expandableListView: ExpandableListView
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+        val view = inflater.inflate(R.layout.faq_fragment, container, false)
+        expandableListView = view.findViewById<View>(R.id.faqlist) as ExpandableListView
+        fetchDataFromFirebase()
+        return view
+    }
+
+    private fun fetchDataFromFirebase() {
+        firebaseHelper.faqDatabase.addListenerForSingleValueEvent(object : ValueEventListener {
+            override fun onDataChange(dataSnapshot: DataSnapshot) {
+                val rows = ArrayList<FaqEvent>()
+                for (faqSnapshot in dataSnapshot.children) {
+                    val data = faqSnapshot.getValue(FaqEvent::class.java)
+                    if (data != null) {
+                        rows.add(data)
+                    }
+                }
+
+                expandableListView.setAdapter(FaqExpandable(rows.toList()))
+            }
+
+            override fun onCancelled(databaseError: DatabaseError) {
+                Log.e(javaClass.canonicalName, "onCancelled", databaseError.toException())
+            }
+        })
+
+    }
+
+    inner class FaqExpandable(faqData: List<FaqEvent>) : BaseExpandableListAdapter() {
+
+        val questionsGroup: List<String?>
```

Comment:
should be List<String>...the questions are non-null

Reply:
done